src/Microsoft.AspNet.Hosting/WebHostOptions.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebHostOptions
+    {
+        private const string ApplicationKey = "app";
+        private const string DetailedErrorsKey = "detailederrors";
+        private const string EnvironmentKey = "environment";
+        private const string OldEnvironmentKey = "ENV";
+        private const string ServerKey = "server";
+        private const string WebRootKey = "webroot";
+
+        public WebHostOptions()
+        {
+        }
+
+        public WebHostOptions(IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            Application = configuration[ApplicationKey];
+            DetailedErrors = string.Equals("true", configuration[DetailedErrorsKey], StringComparison.OrdinalIgnoreCase)
+                || string.Equals("1", configuration[DetailedErrorsKey], StringComparison.OrdinalIgnoreCase);
+            Environment = configuration[EnvironmentKey] ?? configuration[OldEnvironmentKey];
+            Server = configuration[ServerKey];
+            WebRoot = configuration[WebRootKey];
```

Comment:
The upside, is you can nuke the keys (the property names for the POCO are automatically the keys.  It would just work except for the weirdness around DetailedErrors and supporting the old Env key...